examples/dtls-sock/dtls-client.c

```diff
@@ -68,18 +76,102 @@ static const credman_credential_t credential = {
 };
 #endif
 
-static int client_send(char *addr_str, char *data, size_t datalen)
+static void _timeout_handler(event_t *);
+
+static uint8_t _recv_buf[512];
+static sock_udp_t _udp_sock;
+static sock_dtls_t _dtls_sock;
+static event_timeout_t _timeouter;
+static event_t _timeout = { .handler = _timeout_handler };
+static bool _sending = false;
+
+static void _timeout_handler(event_t *event)
+{
+    (void)event;
+    puts("Session handshake timed out");
+    sock_dtls_close(&_dtls_sock);
+    sock_udp_close(&_udp_sock);
+    _sending = false;
```

Comment:
```suggestion
    _close_sock(&_dtls_sock);
```

Reply:
I was thinking about that, but the semantics of this is different. `_close_sock()` is for when the UDP `sock` is not available, which isn't the case here. Also: I'm not even sure we keep this (see commit message, also it would make much more sense from the usual flow to make the server asynchronous). So please focus your review on the `sock_async` integration first and then we can discuss the application.

Reply:
> So please focus your review on the `sock_async` integration first and then we can discuss the application.

At a first glance it looks good, but as always some more comments would be appreciated.